packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -258,4 +273,142 @@ describe('Select', () => {
+            const currentSelection = pageObject.getSelectedOption();
+            expect(currentSelection.text).toBe('One');
+            expect(element.value).toBe('one');
+
+            await pageObject.openAndSetFilterText('T'); // Matches 'Two' and 'Three'
+            pageObject.pressEnter();
+            expect(element.value).toBe('two'); // 'Two' is first option in list so it should be selected now
+            expect(element.open).toBeFalse();
+        });
+
+        it('filtering out current selected item and then clicking selected option changes value and closes popup', async () => {
+            const currentSelection = pageObject.getSelectedOption();
+            expect(currentSelection.text).toBe('One');
+            expect(element.value).toBe('one');
+
+            await pageObject.openAndSetFilterText('T'); // Matches 'Two' and 'Three'
+            pageObject.clickSelectedItem();
+            expect(element.value).toBe('two'); // 'Two' is first option in list so it should be selected now
+            expect(element.open).toBeFalse();
+        });
+
+        it('filtering out current selected item and then clicking non-selected option changes value and closes popup', async () => {
+            const currentSelection = pageObject.getSelectedOption();
+            expect(currentSelection.text).toBe('One');
+            expect(element.value).toBe('one');
+
+            await pageObject.openAndSetFilterText('T'); // Matches 'Two' and 'Three'
+            pageObject.clickOption(1); // index 1 matches option with 'Three' text
+            expect(element.value).toBe('three');
+            expect(element.open).toBeFalse();
+        });
+
+        it('filtering out current selected item and then losing focus changes value and closes popup', async () => {
+            const currentSelection = pageObject.getSelectedOption();
+            expect(currentSelection.text).toBe('One');
+            expect(element.value).toBe('one');
+
+            await pageObject.openAndSetFilterText('T'); // Matches 'Two' and 'Three'
+            await pageObject.clickAway();
+            expect(element.value).toBe('two'); // 'Two' is first option in list so it should be selected now
+            expect(element.open).toBeFalse();
+        });
+
+        it('allows <Space> to be used as part of filter text', async () => {
+            await pageObject.openAndSetFilterText(' '); // Matches 'Has Space'
+            const currentSelection = pageObject.getSelectedOption();
+            expect(currentSelection.text).toBe('Has Space');
+            expect(element.open).toBeTrue();
+        });
+
+        it('opening dropdown after applying filter previously starts with empty filter', async () => {
+            await pageObject.openAndSetFilterText('T'); // Matches 'Two' and 'Three'
+            await pageObject.closeDropdown();
+            await pageObject.clickSelect();
+
+            expect(pageObject.getFilteredOptions().length).toBe(5);
```

Comment:
Can this test also check that the filter input is blank, not just that the options aren't filtered?